xbmc/utils/DisplayInfo.cpp

```diff
@@ -0,0 +1,170 @@
+/*
+ *  Copyright (C) 2023 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "DisplayInfo.h"
+
+#include "utils/log.h"
+
+extern "C"
+{
+#include <libdisplay-info/cta.h>
+#include <libdisplay-info/edid.h>
+#include <libdisplay-info/info.h>
+}
+
+using namespace KODI;
+using namespace UTILS;
+
+void CDisplayInfo::DiInfoDeleter::operator()(di_info* p)
+{
+  di_info_destroy(p);
+}
+
+std::unique_ptr<CDisplayInfo> CDisplayInfo::Create(std::vector<uint8_t> edid)
+{
+  struct DisplayInfoMaker : public CDisplayInfo
+  {
+    DisplayInfoMaker(std::vector<uint8_t> edid) : CDisplayInfo(edid) {}
+  };
+
+  auto info = std::make_unique<DisplayInfoMaker>(edid);
```

Comment:
```suggestion
std::unique_ptr<CDisplayInfo> info(new CDisplayInfo(edid));
```
I probably wouldn't do the inheritance trick without the base class destructor being virtual. Nothing bad **should** happen because `DisplayInfoMaker`'s destructor doesn't do any work but is it defined behaviour?

Reply:
std::optional may be a good pattern here, though I think it requires a fully defined type so no forward declarations can be made.

Reply:
I couldn't get `std::optional` to work here, I'm not exactly sure why but I didn't want to spend more time on it.

```
/home/lukas/Documents/git/xbmc/xbmc/windowing/gbm/WinSystemGbm.cpp:117:48: error: use of deleted function ‘std::optional<KODI::UTILS::CDisplayInfo>& std::optional<KODI::UTILS::CDisplayInfo>::operator=(std::optional<KODI::UTILS::CDisplayInfo>&&)’
  117 |       m_info = UTILS::CDisplayInfo::Create(edid);
      |                                                ^
In file included from /home/lukas/Documents/git/xbmc/xbmc/utils/DisplayInfo.h:12,
                 from /home/lukas/Documents/git/xbmc/xbmc/windowing/gbm/WinSystemGbm.h:16,
                 from /home/lukas/Documents/git/xbmc/xbmc/windowing/gbm/WinSystemGbm.cpp:9:
/usr/include/c++/12/optional:705:11: note: ‘std::optional<KODI::UTILS::CDisplayInfo>& std::optional<KODI::UTILS::CDisplayInfo>::operator=(std::optional<KODI::UTILS::CDisplayInfo>&&)’ is implicitly deleted because the default definition would be ill-formed:
  705 |     class optional
      |           ^~~~~~~~
/usr/include/c++/12/optional:705:11: error: use of deleted function ‘std::_Enable_copy_move<false, false, false, false, _Tag>& std::_Enable_copy_move<false, false, false, false, _Tag>::operator=(std::_Enable_copy_move<false, false, false, false, _Tag>&&) [with _Tag = std::optional<KODI::UTILS::CDisplayInfo>]’
In file included from /usr/include/c++/12/bits/hashtable.h:36,
                 from /usr/include/c++/12/unordered_map:46,
                 from /usr/include/c++/12/functional:61,
                 from /usr/include/spdlog/common.h:16,
                 from /usr/include/spdlog/spdlog.h:12,
                 from /home/lukas/Documents/git/xbmc/xbmc/utils/log.h:36,
                 from /home/lukas/Documents/git/xbmc/xbmc/threads/SystemClock.h:11,
                 from /home/lukas/Documents/git/xbmc/xbmc/windowing/gbm/WinSystemGbm.h:14:
/usr/include/c++/12/bits/enable_special_members.h:309:5: note: declared here
  309 |     operator=(_Enable_copy_move&&) noexcept                         = delete;
      |     ^~~~~~~~

```